### ISSUE TITLE
Consuming compiled Typescript lib with side effect (CDN vs ns require)

### ISSUE THREAD

**Issue by rlhk** [thheller/shadow-cljs#1061]
I made a Logseq plugin: https://github.com/rlhk/logseq-url-plus with shadow-cljs by loading the plugin lib from CDN, which works well with advanced compilation.

That said, being a bit paranoid I'm trying to figure out how to use the lib from ns require like: `(ns (:require ... ["@logseq/libs"] ...)` which always gives the error: `ReferenceError: Must call super constructor in derived class before accessing 'this' or returning from derived constructor`

The following is a capture of importing the .js file directly from class path, which should be the same as ns require
<img width="978" alt="Screenshot 2022-11-08 at 14 17 55" src="https://user-images.githubusercontent.com/17818753/200489510-e89f9531-7406-4ec6-87f0-68e12b27ec51.png">

Basically I understand such imported .js is wrapped by shadow-cljs as: `shadow$provide.module$lsplugin_user = function(global, require, module, exports) { ...` where the `...` part is the compiled js code (same as https://cdn.jsdelivr.net/npm/@logseq/libs)

So the question is could anything be done to the ns require to make it work like how other folks use it from .ts projects? Hopefully to take advantage of further code size optimization?

`import "@logseq/libs";`

Or loading in index.html from CDN is legit enough?

**Comment by thheller** [comment]
I do not understand the question.

I do not know how the logseq plugin system works or what it expects. I don't know "how other folks use it from .ts projects", so I really can't say what the problem even is here.

**Comment by rlhk** [comment]
@thheller Thanks for your quick response.

From my understanding, I think logseq provides a global object `logseq` (Electron App), the plugin lib from CDN just load and run and returns `true` after evaluation, and attach a bunch of objects and methods on it for plugin's consumption.

I guess the question is why loading the bare compiled js works but not the shadow-cljs wrapped way.

**Comment by thheller** [comment]
Sorry, I can't really help you debug this. I have no not logseq and I do not know how to use it.

From a brief look the cdn version is different than the npm version. Maybe just a version problem?

**Comment by rlhk** [comment]
No worries, will see if a different version works. 
By the way, the Logseq team is looking at the issue, will report back here.
Thanks.

**Comment by tiensonqin** [comment]
The related typescript source code is at https://github.com/logseq/logseq/blob/master/libs/src/LSPlugin.caller.ts#L46.

**Comment by thheller** [comment]
The Closure Compiler is doing this. Not something I can control from the shadow-cljs side of things. Looks like a bug to me.

**Comment by rlhk** [comment]
> Btw, you don't need to download Logseq to test https://github.com/rlhk/logseq-url-plus, `yarn dev` and open `http://localhost:8080` should be enough to inspect the errors.

To clarify, no such error on my production code as the lib is loaded from CDN in entry index.html.

To reproduce the error:
1. Remove the script loading from index.html
2. Add `["@logseq/libs"]` in the ns require of core.cljs.

**Comment by rlhk** [comment]
Issue resolved as of [@logseq/libs](https://www.npmjs.com/package/@logseq/libs) 0.0.12 & [logseq-url-plus 0.0.9](https://github.com/rlhk/logseq-url-plus/releases/tag/0.0.9)
